pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -42,6 +43,28 @@ func validName(ctx Context) validation.Result {
 	return validation.OK
 }
 
+func specUpdatedToBeta(ctx Context) validation.Result {
+	oldAPIVersion := "elasticsearch.k8s.elastic.co/v1alpha1"
+
+	es := ctx.Proposed.Elasticsearch
+	if es.APIVersion == oldAPIVersion {
+		return validation.Result{Reason: fmt.Sprintf("%s: outdated APIVersion", validationFailedMsg)}
+	}
+
+	if len(es.Spec.NodeSets) == 0 {
```

Comment:
nit: Is it not redundant with https://github.com/elastic/cloud-on-k8s/blob/82020d17bd8a3f8583222eca437258903590c830/pkg/controller/elasticsearch/validation/validations.go#L80 ?

Reply:
It is, but the messaging is different. Here we will communicate what the issue is and mention `nodeSet` specifically. If someone is still using `nodes` and sees the reason that `hasMaster` provides, it's probably confusing (it was for me).

Reply:
My idea was that the original message could have been improved :)
But I'm 👍with you changes